From 102abaf85a87d2acfbca0a2ff0d639243afd4f71 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 12 Oct 2017 14:28:49 +0800
Subject: [PATCH 2623/5242] MLK-16574 gpu: imx: dpu: layerblend: Fix
 layerblend_shdldsel()

commit  af146d8bec538682cb827ce6efbe99afd3630bb1 from
https://source.codeaurora.org/external/imx/linux-imx.git

The shadow load selection field of layerblend is at bit[2:1].
Thus, we need to shift to left by one bit for the selection value.

Fixes: 4ec895ea31a2 ("MLK-15001-11 gpu: Move ipuv3 and dpu to imx folder")
Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-layerblend.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-layerblend.c b/drivers/gpu/imx/dpu/dpu-layerblend.c
index c7c42f3..bc78299 100644
--- a/drivers/gpu/imx/dpu/dpu-layerblend.c
+++ b/drivers/gpu/imx/dpu/dpu-layerblend.c
@@ -52,6 +52,7 @@
 #define SHDTOKSEL				(0x3 << 3)
 #define SHDTOKSEL_SHIFT				3
 #define SHDLDSEL				(0x3 << 1)
+#define SHDLDSEL_SHIFT				1
 #define CONTROL					0xC
 #define MODE_MASK				BIT(0)
 #define BLENDCONTROL				0x10
@@ -213,7 +214,7 @@ void layerblend_shdldsel(struct dpu_layerblend *lb, lb_shadow_sel_t sel)
 	mutex_lock(&lb->mutex);
 	val = dpu_lb_read(lb, STATICCONTROL);
 	val &= ~SHDLDSEL;
-	val |= sel;
+	val |= (sel << SHDLDSEL_SHIFT);
 	dpu_lb_write(lb, val, STATICCONTROL);
 	mutex_unlock(&lb->mutex);
 }
-- 
1.7.9.5

